x/delegation/keeper/delegation.go

```diff
@@ -136,3 +148,107 @@ func (k *Keeper) UndelegateFrom(ctx sdk.Context, params *delegationtype.Delegati
 	// call the hooks registered by the other modules
 	return k.Hooks().AfterUndelegationStarted(ctx, params.OperatorAddress, delegationtype.GetUndelegationRecordKey(r.BlockNumber, r.LzTxNonce, r.TxHash, r.OperatorAddr))
 }
+
+func (k *Keeper) MarkSelfDelegatedOperator(
+	ctx sdk.Context,
+	clientChainID uint64,
+	operatorAddress sdk.AccAddress,
+	stakerAddress common.Address,
+) error {
+	if !k.operatorKeeper.IsOperator(ctx, operatorAddress) {
+		return delegationtype.ErrOperatorNotExist
+	}
+
+	stakerID, _ := assetstype.GetStakeIDAndAssetID(clientChainID, stakerAddress[:], nil)
+	oldOperator, err := k.GetSelfDelegatedOperator(ctx, stakerID)
+	if err != nil {
+		return err
+	}
+
+	if oldOperator != "" && oldOperator == operatorAddress.String() {
```

Comment:
I don't think we should support reassigning a staker's self delegation to another operator.

Reply:
Yeah, it's fine not to support reassigning a self-delegation. The staker can correct the marking mistake by unmarking and marking the operator again. I will remove the reassigning logic and check if there has been a marked operator before marking.